[#108968418] add logsearch kibana and proxy

mtekel opened the pull request:
[Story](https://www.pivotaltracker.com/n/projects/1275640/stories/108968418)
### What

Add the original, unmodified kibana to logsearch. To make access to this kibana easy, deploy proxy app to CF which adds basic auth on top.
### Testing

Apply to existing env. or build from scratch. Verify you can access kibana on `logs.<environment domain>`. The password is same as the UAA enabled Kibana secret (see pass secrets or secrets.yml on bastion). User is kibana.
### Who

Anyone but @mtekel.

mtekel committed:
Bastion: add apache2-utils package

This provides htpasswd tool used for generating basic auth password
file for nginx based on password we store in our secrets.

mtekel committed:
Deploy kibana proxy to CF

Deploy a basic auth enabled proxy for kibana into CF. This is in order
to make kibana more easily accessible (otherwise you have to make SSH
tunnels via bastion and add kibana into your hosts file). The basic
auth prevents unauthorized access. The deploy script also creates the
necessary security group to let the proxy reach kibana server.

This is deployed as a replacement for the UAA enabled kibana that has
disk read issues. Once we get that app working we can revert adding this
proxy and the kibana on the logsearch servers instead. The code of the
deploy script could be DRYed with other 2 CF app deployment scripts
(graphite nozzle and psql), but this is just a temporary workaround on
a legacy build, so no further effort to improve things was spent.

combor commented:
Deployment failed with:

```
API endpoint:   https://api.piotr.cf.paas.alphagov.co.uk (API version: 2.34.0)
User:           admin
Org:            admin
Space:          admin
Usage:
        htpasswd [-cimBdpsDv] [-C cost] passwordfile username
        htpasswd -b[cmBdpsDv] [-C cost] passwordfile username password

        htpasswd -n[imBdps] [-C cost] username
        htpasswd -nb[mBdps] [-C cost] username password
 -c  Create a new file.
 -n  Don't update file; display results on stdout.
 -b  Use the password from the command line rather than prompting for it.
 -i  Read password from stdin without verification (for script usage).
 -m  Force MD5 encryption of the password (default).
 -B  Force bcrypt encryption of the password (very secure).
 -C  Set the computing time used for the bcrypt algorithm
     (higher is more secure but slower, default: 5, valid: 4 to 31).
 -d  Force CRYPT encryption of the password (8 chars max, insecure).
 -s  Force SHA encryption of the password (insecure).
 -p  Do not encrypt the password (plaintext, insecure).
 -D  Delete the specified user.
 -v  Verify password for the specified user.
On other systems than Windows and NetWare the '-p' flag will probably not work.
The SHA algorithm does not use a salt and is less secure than the MD5 algorithm.
```
and

```
FAILED
Server error, status code: 400, error code: 300001, message: The security group is invalid: rules rule number 1 contains invalid destination
FAILED
security group kibana-proxy not found
FAILED
security group kibana-proxy not found
```

and 

```
Starting app logs in org admin / space admin as admin...
-----> Downloaded app package (4.0K)
-------> Buildpack version 1.2.1
grep: Staticfile: No such file or directory
-----> Using root folder
-----> Copying project files into public/
-----> Setting up nginx
grep: Staticfile: No such file or directory
-----> Uploading droplet (2.5M)

0 of 1 instances running, 1 down
0 of 1 instances running, 1 down
0 of 1 instances running, 1 down
0 of 1 instances running, 1 starting
0 of 1 instances running, 1 down
0 of 1 instances running, 1 down
0 of 1 instances running, 1 down
0 of 1 instances running, 1 down
0 of 1 instances running, 1 starting
0 of 1 instances running, 1 down
0 of 1 instances running, 1 down
0 of 1 instances running, 1 down
0 of 1 instances running, 1 failing
FAILED
Start unsuccessful
```

mtekel commented:
This failed because `elasticsearch_ip=$(bosh vms 2> /dev/null | grep logsearch_minimal | awk '{print $8}')` ended up as empty value. Not sure why, as it works the 2nd time. Maybe there's some race conditions just after CF deployment...

mtekel committed:
Add kibana job to logsearch manifest

Use the kibana job from the logsearch release and deploy it on the same
server that hosts all other logsearch jobs. Configure the elasticsearch
endpoint for kibana.

mtekel commented:
I have updated the scripts to get elasticseach IP in other way (directly from final logsearch manifest), which should work even if there's still some work in progress by BOSH, preventing it from returning the IP.

combor commented on scripts/deploy_kibana_proxy.sh:
```diff
@@ -0,0 +1,67 @@
+#!/bin/bash
+if [[ -n $(cf app logs | grep running) ]] ; then
```
Minor thing but the -n test requires that the string be quoted within the test brackets. Using an unquoted string is an unsafe practice.  Examples:
http://www.tldp.org/LDP/abs/html/comparison-ops.html#STRTEST

combor replied:
We found that "[[" fixes this problem. Merging....
